resources/analysisTools/snvPipeline/in_dbSNPcounter.pl

```diff
@@ -37,85 +37,80 @@
 
 
 my $header = "";
-while ($header = <FH>)
-{
-	last if ($header =~ /^\#CHROM/); # that is the line with the column names
-}
-chomp $header;
-
-my $DBSBP = 13;
-my $KG = 14;
-my $CONFIDENCE = 28;
-my $CANNOTATION = -1;
-my $RECLASSIFICATION = -1;
-my $CLASSIFICATION;
+my ($DBSBP, $KG, $CONFIDENCE, $CLASSIFICATION);
 
-@help = split(/\t/, $header);
-for (my $c = 0; $c < @help; $c++)
-{
-	# fixed column labels (from vcf_pileup_compare)
-	if ($help[$c] eq "CONFIDENCE")
-	{
-		$CONFIDENCE = $c;
-		print STDERR "CONFIDENCE in column $c\n";
-	}
-	if ($help[$c] eq "1K_GENOMES")
+while(!eof($FH)){
+	my $line = readline($FH) || die "Could not read line: $?";
+	chomp $line;
+	if ($line =~ /^#/)
 	{
-		$KG = $c;
-		print STDERR "INFO in column $c\n";
-	}
-	if ($help[$c] eq "DBSNP")
-	{
-		$DBSBP = $c;
-		print STDERR "DBSNP_COL in column $c\n";
-	}
-	if ($help[$c] eq "ANNOTATION_control")
-    {
-    	$CANNOTATION = $c;
-    	print STDERR "ANNOTATION_control in column $c\n";
-    }
-    if ($help[$c] eq "RECLASSIFICATION")
-    {
-        $RECLASSIFICATION = $c;
-        print STDERR "RECLASSIFICATION in column $c\n";
-    }
-}
+		#print STDOUT "HELLO\n";
+		if ($line =~/^#CHROM/){
+			chomp $line;
 
-if ($CANNOTATION > 0) {
-    $CLASSIFICATION = $CANNOTATION;
-} else {
-    $CLASSIFICATION = $RECLASSIFICATION;
-}
+			my @head=split("\t", $line);
+			my %col;
 
-# I know that it's evilly hardcoded!
-while (<FH>)
-{
-	if ($_ =~ /^#/)	# skip header lines
-	{
-		next;
-	}
-	$all++;
-	@help = split ("\t", $_);
-	if ($help[$CLASSIFICATION] =~ /somatic/)
-	{
-		$somatic++;
-		if ($help[$CONFIDENCE] >= $minscore)
-		{
-			$scored++;
-			#print STDERR $_;
-			if ($help[$DBSBP] =~ /MATCH=exact/)
+			my $i = 0;
+			while($i < @head)
```

Comment:
Not sure. I try to avoid this and always use explicit cast
```suggestion
			while($i < scalar(@head))
```